craft_application/models/project.py

```diff
@@ -113,3 +129,18 @@ def transform_pydantic_error(cls, error: pydantic.ValidationError) -> None:
                 # "input" key in the error dict, so we can't put the original
                 # value in the error message.
                 error_dict["msg"] = message
+
+
+# This is a hack to allow us to use the same type for the project model
```

Comment:
I don't think this is a good idea! In a way it's the worst of both worlds, because it'll satisfy the typing but might fail in runtime because there's no guarantee that a project has the mixin as a superclass.

The way I see it, we should keep the mixin separate and revert to the original `Project`, and then the Application class can check whether the specific project class (available in the metadata) is a subclass of the mixin (via `issubclass`). This has to be done at runtime, unfortunately.